src/meta/service/src/ddl_service.rs

```diff
@@ -119,6 +122,96 @@ impl DdlServiceImpl {
             fragment_graph: fragment_graph.unwrap(),
         }
     }
+
+    pub fn start_migrate_table_fragments(&self) -> (JoinHandle<()>, Sender<()>) {
+        tracing::info!("start migrate legacy table fragments task");
+        let env = self.env.clone();
+        let metadata_manager = self.metadata_manager.clone();
+        let ddl_controller = self.ddl_controller.clone();
+
+        let (shutdown_tx, mut shutdown_rx) = tokio::sync::oneshot::channel();
+        let join_handle = tokio::spawn(async move {
+            async fn migrate_inner(
+                shutdown_rx: &mut tokio::sync::oneshot::Receiver<()>,
+                env: &MetaSrvEnv,
+                metadata_manager: &MetadataManager,
+                ddl_controller: &DdlController,
+            ) -> MetaResult<()> {
+                let tables = metadata_manager
+                    .catalog_controller
+                    .list_unmigrated_tables()
+                    .await?;
+
+                if tables.is_empty() {
+                    return Ok(());
+                }
+
+                let client = {
+                    let workers = metadata_manager
+                        .list_active_worker_node(Some(WorkerType::Frontend))
+                        .await?;
+                    if workers.is_empty() {
+                        return Err(anyhow::anyhow!("no active frontend nodes found").into());
+                    }
+                    let worker = workers.choose(&mut thread_rng()).unwrap().clone();
+                    env.frontend_client_pool().get(&worker).await?
+                };
+
+                for table in tables {
+                    match shutdown_rx.try_recv() {
+                        Ok(_) | Err(tokio::sync::oneshot::error::TryRecvError::Closed) => {
+                            tracing::info!("shutting down migrate table fragments task");
+                            return Ok(());
+                        }
+                        Err(tokio::sync::oneshot::error::TryRecvError::Empty) => {}
+                    }
+
+                    let req = GetTableReplacePlanRequest {
+                        database_id: table.database_id,
+                        owner: table.owner,
+                        table_name: table.name.clone(),
+                        cdc_table_change: None,
+                    };
+                    let resp = client
+                        .get_table_replace_plan(req)
+                        .await
+                        .context("failed to get table replace plan from frontend")?;
+
+                    let plan = resp.into_inner().replace_plan.unwrap();
+                    let replace_info = DdlServiceImpl::extract_replace_table_info(plan);
+                    ddl_controller
+                        .run_command(DdlCommand::ReplaceStreamJob(replace_info))
+                        .await?;
+                    tracing::info!("migrated table fragments for table {}", table.name);
```

Comment:
May include the `table_id` in the log in case of table name duplication, and also the time taken to migrate the table.